subprojects/parseq-restli-client/src/main/java/com/linkedin/restli/client/ParSeqRestClient.java

```diff
@@ -289,7 +266,7 @@ private boolean needApplyTaskTimeout(RequestContext requestContext, ConfigValue<
     if (!taskNeedTimeout) {
       return requestTask;
     } else {
-      return withD2Timeout(requestTask, timeout);
+      return withTimeout(requestTask, timeout);
```

Comment:
Put a comment here 
// still enforce parseq client timeout if for some reason downstream services are not timed out properly.

Reply:
Also, would you mind make one javadoc change in method 
```
  // check whether we need to apply timeout to a rest.li request task.
 private boolean needApplyTaskTimeout(RequestContext requestContext, ConfigValue<Long> timeout) {
    // **return false** if no timeout configured or per-request timeout already specified in request context
     return timeout.getValue() != null && timeout.getValue() > 0 && !hasRequestContextTimeout(requestContext);
  }
```

Reply:
Update CHANGELOG.md by following our developer guide.

Reply:
Updated